### DISCUSSION TITLE
unable to start new install

### DISCUSSION THREAD

**Discussion by rsuinux** [mastodon/mastodon#27843] [General]
hello all,
I'm trying to start a first installation of mastodon via docker.
By following the Readme, I execute the various commands:

`docker compose -f .devcontainer/docker-compose.yml up -d`

OK for now.
But the following fails:
```

docker compose -f .devcontainer/docker-compose.yml exec app .devcontainer/post-create.sh
OCI runtime exec failed: exec failed: unable to start container process: exec: ".devcontainer/post-create.sh": stat .devcontai
ner/post-create.sh: no such file or directory: unknown

```
post-create.sh doesn't seem to be known (which it isn't). However, I can of course edit it (vi, less, ...)

I must have missed something?

Do you have a suggestion?
Thanks.
Rémi.

**Comment by vmstan** [comment]
This is a known issue the team is working on.

**Comment by rsuinux** [comment]
Thank you for this feedback. Is there a workaround, or should this step be avoided?

**Reply by vmstan** [reply]
#27678 is latest

**Reply by dginovker** [reply]
Seems to be a different issue, even with https://github.com/mastodon/mastodon/pull/28099 merged, I'm getting the same error as OP

(Simply followed the Docker steps on a fresh Arch Linux install)

**Comment by juliusrajala** [comment]
I'm seeing this same issue on MacOS at the moment. Is there any update on this, or should running the development environment via Docker be avoided for now?

**Reply by juliusrajala** [reply]
Investigation revealed this is a simple case of a missing workdir -setup at least on local Docker. The `post-create.sh` and Procfile are run from the root of the docker container. Executing these with a `-w /workspaces/mastodon` before the container's tag starts the scripts correctly. 

I also noticed that the docker-compose iinside the devcontainer is a bit greedy grabbing pushing all cached workspaces in the parent directory into the container. This isn't a big issue but it's unnecessary.

**Comment by alexandreumb** [comment]
hi is there any solution to this problem already, i am trying to do the installation but i have the problem still.
![image](https://github.com/mastodon/mastodon/assets/106603120/12fdefdb-f841-4e97-9d5a-7d1d05055630)

**Reply by juliusrajala** [reply]
https://github.com/mastodon/mastodon/pull/28402/files you can apply this fix manually. That should help.

**Reply by alexandreumb** [reply]
Thank you it helped but, i am now getting this issue
![image](https://github.com/mastodon/mastodon/assets/106603120/cfd22260-6607-48f9-bff9-0defa68a71d3)
and i already run the command git config --config --all safe.directory /workspaces/mastodon, and i am still getting that.

**Reply by MJRT** [reply]
same issue

**Reply by DPIDNB** [reply]
I have this as well

**Reply by Osama-Yusuf** [reply]
execute the command it says inside the container as follows 
sudo docker-compose -f .devcontainer/docker-compose.yml exec app git config --global --add safe.directory /workspaces/mastodon
sudo docker-compose -f .devcontainer/docker-compose.yml exec app .devcontainer/post-create.sh
